Source: Dimitry Ishenko <dimitry.ishenko@gmail.com>
Upstream: submitted via IRC
Reason: Currently systemd uses -D option in order to run gpm in foreground,
which spews all kinds of debugging crap into the log

diff -ru gpm-1.20.6.orig/src/daemon/cmdline.c gpm-1.20.6/src/daemon/cmdline.c
--- gpm-1.20.6.orig/src/daemon/cmdline.c	2011-09-22 19:25:58.280107991 -0400
+++ gpm-1.20.6/src/daemon/cmdline.c	2011-09-22 19:41:58.763444491 -0400
@@ -32,7 +32,7 @@
 void cmdline(int argc, char **argv)
 {
    extern struct options option;
-   char options[]="a:A::b:B:d:Dg:hi:kl:m:Mo:pr:R::s:S:t:Tuv23";
+   char options[]="a:A::b:B:d:DFg:hi:kl:m:Mo:pr:R::s:S:t:Tuv23";
    int  opt;
 
    /* initialize for the dual mouse */ 
@@ -49,6 +49,7 @@
          case 'B': (which_mouse->opt_sequence) = optarg;                break;
          case 'd': (which_mouse->opt_delta) = atoi(optarg);             break;
          case 'D': option.run_status = GPM_RUN_DEBUG;    break;
+         case 'F': option.run_status = GPM_RUN_FOREGROUND;              break;
          case 'g': (which_mouse->opt_glidepoint_tap)=atoi(optarg);      break;
          case 'h': exit(usage(NULL));
          case 'i': (which_mouse->opt_time)=atoi(optarg);                break;
diff -ru gpm-1.20.6.orig/src/headers/message.h gpm-1.20.6/src/headers/message.h
--- gpm-1.20.6.orig/src/headers/message.h	2011-09-22 19:25:58.240107992 -0400
+++ gpm-1.20.6/src/headers/message.h	2011-09-22 19:46:43.650112096 -0400
@@ -58,6 +58,7 @@
 #define GPM_RUN_STARTUP 0
 #define GPM_RUN_DAEMON  1
 #define GPM_RUN_DEBUG   2
+#define GPM_RUN_FOREGROUND 3
 
 /* messages */
 
@@ -77,6 +78,7 @@
          "    -B sequence      allows changing the buttons (default '%s')\n" \
          "    -d delta         sets the delta value (default %d) (must be 2 or more)\n" \
          "    -D	             debug mode: don't auto-background\n" \
+         "    -F	             run in foreground\n" \
          "    -g tap-button    sets the button (1-3) that is emulated by tapping on\n" \
          "                     a glidepoint mouse, none by default. (mman/ps2 only)\n" \
          "    -i interval      max time interval for multiple clicks (default %i)\n" \
diff -ru gpm-1.20.6.orig/src/report.c gpm-1.20.6/src/report.c
--- gpm-1.20.6.orig/src/report.c	2011-09-22 19:25:58.236774658 -0400
+++ gpm-1.20.6/src/report.c	2011-09-22 19:41:00.326777632 -0400
@@ -135,6 +135,7 @@
          break; /* startup sequence */   
 
       /******************** RUNNING *****************/
+      case GPM_RUN_FOREGROUND:
       case GPM_RUN_DAEMON:
          switch(stat) {
             case GPM_STAT_INFO:
